aten/src/ATen/ParallelOpenMP.cpp

```diff
@@ -37,6 +37,11 @@ void init_num_threads() {
     // size of the OpenMP thread pool, resulting in worse performance (and memory
     // leaks in GCC 5.4)
     omp_set_num_threads(mkl_get_max_threads());
+    // Dynamic adjustment of the number of threads by MKL is enabled by default.
+    // So, MKL may use fewer threads in parallel regions than the number returned by mkl_get_max_threads.
+    // With libgomp (GCC's OpenMP), this may result in the creation & destruction of threads across every
+    // MKL & non-MKL boundary of OpenMP usage, resulting in degraded performance.
```

Comment:
I don't understand why there would be "creation & destruction" cost if MKL uses less number of OMP threads. OMP runtime should maintain a pool of threads and don't have to "create/destroy" threads for every OMP region?